### ISSUE TITLE
Build failed on optimize=1 or 2 with pystray imported

### ISSUE THREAD

**Issue by MaxBQb** [py2exe/py2exe#120]
Python: 3.9.5
py2exe: 0.11.0.1

```py
import os.path
from distutils.core import setup
from glob import glob
import py2exe
import _meta as app


setup(console=[dict(
        script='main.py',
        icon_resources=[(0, app.__icon__)],
      )],
      name=app.__product_name__,
      version=app.__version__,
      author=app.__author__,
      data_files=[('.', ["update.bat"]),
                  ('./img', glob('img/*'))],
      options=dict(
          py2exe=dict(
              optimize=1
          )
      ))
```

While this code works perfect on py2exe 0.10.4.0
On the latest version I've this:
```
C:\Users\MyUser\PycharmProjects\MyProject\venv\Scripts\python.exe C:/Users/MyUser/PycharmProjects/MyProject/setup.py py2exe
running py2exe
Traceback (most recent call last):
  File "C:\Users\MyUser\PycharmProjects\MyProject\setup.py", line 8, in <module>
    setup(console=[dict(
  File "C:\Program Files\Python39\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Program Files\Python39\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Program Files\Python39\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\distutils_buildexe.py", line 192, in run
    self._run()
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\distutils_buildexe.py", line 272, in _run
    builder.analyze()
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\runtime.py", line 177, in analyze
    target.analyze(mf)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\runtime.py", line 78, in analyze
    modulefinder.run_script(self.script)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 197, in run_script
    self.load_module('__main__', fp, pathname, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 402, in load_module
    self.scan_code(co, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 475, in scan_code
    self._safe_import_hook(name, m, fromlist, level=0)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 420, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 211, in import_hook
    q, tail = self.find_head_package(parent, name)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 267, in find_head_package
    q = self.import_module(head, qname, parent)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 361, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 402, in load_module
    self.scan_code(co, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 475, in scan_code
    self._safe_import_hook(name, m, fromlist, level=0)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 420, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 212, in import_hook
    m = self.load_tail(q, tail)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 289, in load_tail
    m = self.import_module(head, mname, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 361, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 402, in load_module
    self.scan_code(co, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 475, in scan_code
    self._safe_import_hook(name, m, fromlist, level=0)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 420, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 211, in import_hook
    q, tail = self.find_head_package(parent, name)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 267, in find_head_package
    q = self.import_module(head, qname, parent)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 361, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 374, in load_module
    m = self.load_package(fqname, pathname)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 526, in load_package
    self.load_module(fqname, fp, buf, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 402, in load_module
    self.scan_code(co, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 498, in scan_code
    self._safe_import_hook(name, m, fromlist, level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 420, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 211, in import_hook
    q, tail = self.find_head_package(parent, name)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 267, in find_head_package
    q = self.import_module(head, qname, parent)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 361, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 402, in load_module
    self.scan_code(co, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 475, in scan_code
    self._safe_import_hook(name, m, fromlist, level=0)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 435, in _safe_import_hook
    self.import_hook(name, caller, [sub], level=level)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 216, in import_hook
    self.ensure_fromlist(m, fromlist)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 306, in ensure_fromlist
    submod = self.import_module(sub, subname, m)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 354, in import_module
    fp, pathname, stuff = self.find_module(partname,
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 555, in find_module
    return _find_module(name, path)
  File "C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 89, in _find_module
    spec = importlib.machinery.PathFinder.find_spec(name, path)
  File "<frozen importlib._bootstrap_external>", line 1414, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1380, in _get_spec
TypeError: 'int' object is not iterable

Process finished with exit code 1
```

But when I set optimize=0 (as well as don't use potimize param at all)

It works :)

```
running py2exe

  98 missing Modules                 
  ------------------                 
? AppKit                              imported from keyboard._darwinkeyboard
? Cookie                              imported from requests.compat
? IPython.display                     imported from PIL.ImageShow, tqdm.notebook
? IPython.html.widgets                imported from tqdm.notebook
? OpenSSL.SSL                         imported from urllib3.contrib.pyopenssl
? OpenSSL.crypto                      imported from urllib3.contrib.pyopenssl
? PIL._imagingagg                     imported from -
? PyQt5.QtCore                        imported from PIL.ImageQt
? PyQt5.QtGui                         imported from PIL.ImageQt
? PyQt6.QtCore                        imported from PIL.ImageQt
? PyQt6.QtGui                         imported from PIL.ImageQt
? PySide2.QtCore                      imported from PIL.ImageQt
? PySide2.QtGui                       imported from PIL.ImageQt
? PySide6.QtCore                      imported from PIL.ImageQt
? PySide6.QtGui                       imported from PIL.ImageQt
? Quartz                              imported from keyboard._darwinkeyboard
? Queue                               imported from keyboard, keyboard._generic, keyboard._nixcommon, urllib3.util.queue
? StringIO                            imported from requests.compat, six, urllib3.packages.six
? __builtin__                         imported from pkg_resources._vendor.pyparsing
? __main__                            imported from bdb, pdb, pkg_resources
? _frozen_importlib                   imported from importlib, importlib.abc, zipimport
? _frozen_importlib_external          imported from importlib, importlib._bootstrap, importlib.abc, zipimport
? _manylinux                          imported from pkg_resources._vendor.packaging.tags
? _posixshmem                         imported from multiprocessing.resource_tracker, multiprocessing.shared_memory
? _watchdog_fsevents                  imported from watchdog.observers.fsevents
? _winreg                             imported from pkg_resources._vendor.appdirs, platform, requests.utils, winregal.winregal
? asyncio.DefaultEventLoopPolicy      imported from -
? asyncio.run                         imported from __main__
? asyncio.to_thread                   imported from active_window_checker, tray.tray
? backports.ssl_match_hostname        imported from urllib3.packages.ssl_match_hostname
? brotli                              imported from urllib3.response, urllib3.util.request
? cffi                                imported from PIL.Image, PIL.ImageTk, PIL.PyAccess
? collections.Callable                imported from requests.compat
? collections.Iterable                imported from pkg_resources._vendor.pyparsing
? collections.Mapping                 imported from requests.compat, urllib3._collections
? collections.MutableMapping          imported from pkg_resources._vendor.pyparsing, requests.compat, urllib3._collections
? colorama                            imported from tqdm.utils
? com.sun                             imported from pkg_resources._vendor.appdirs
? com.sun.jna                         imported from pkg_resources._vendor.appdirs
? com.sun.jna.platform                imported from pkg_resources._vendor.appdirs
? cookielib                           imported from requests.compat
? cryptography                        imported from requests, urllib3.contrib.pyopenssl
? cryptography.hazmat.backends.openssl imported from urllib3.contrib.pyopenssl
? cryptography.hazmat.backends.openssl.x509 imported from urllib3.contrib.pyopenssl
? cryptography.x509                   imported from urllib3.contrib.pyopenssl
? cryptography.x509.extensions        imported from urllib3.contrib.pyopenssl
? defusedxml.ElementTree              imported from PIL.Image
? dummy.Process                       imported from multiprocessing.pool
? dummy_threading                     imported from requests.cookies
? fastnumbers                         imported from natsort.compat.fastnumbers
? icu                                 imported from natsort.compat.locale, natsort.natsort
? ipywidgets                          imported from tqdm.notebook
? java.lang                           imported from platform
? matplotlib                          imported from tqdm.gui
? matplotlib.pyplot                   imported from tqdm.gui
? multiprocessing.RLock               imported from tqdm.std
? numpy                               imported from PIL.Image, PIL.ImageFilter
? olefile                             imported from PIL.FpxImagePlugin, PIL.MicImagePlugin
? ordereddict                         imported from pkg_resources._vendor.pyparsing
? org.python.core                     imported from copy, pickle
? os.path                             imported from auto_update, ctypes._aix, distutils.file_util, gui, os, pkg_resources, pkgutil, py_compile, requests.adapters, sysconfig, tqdm.cli, tracemalloc, unittest, unittest.util, utils, watchdog.events, watchdog.observers.kqueue, watchdog.observers.read_directory_changes
? packages.six.moves                  imported from urllib3.connectionpool, urllib3.util.queue, urllib3.util.response
? packages.six.moves.http_client      imported from urllib3.connection, urllib3.exceptions
? packages.six.moves.urllib.parse     imported from urllib3.poolmanager, urllib3.request
? packages.ssl_match_hostname.CertificateError imported from urllib3.connection, urllib3.connectionpool
? packages.ssl_match_hostname.match_hostname imported from urllib3.connection
? packaging.version                   imported from PIL.ImageFont
? pandas                              imported from tqdm.std
? pandas.core.frame                   imported from tqdm.std
? pandas.core.groupby                 imported from tqdm.std
? pandas.core.groupby.generic         imported from tqdm.std
? pandas.core.groupby.groupby         imported from tqdm.std
? pandas.core.series                  imported from tqdm.std
? pandas.core.window                  imported from tqdm.std
? pandas.core.window.expanding        imported from tqdm.std
? pandas.core.window.rolling          imported from tqdm.std
? pep517                              imported from importlib.metadata
? pkg_resources.extern.appdirs        imported from pkg_resources
? pkg_resources.extern.packaging      imported from pkg_resources
? pkg_resources.extern.pyparsing      imported from pkg_resources._vendor.packaging.markers, pkg_resources._vendor.packaging.requirements
? readline                            imported from cmd, code, pdb
? resource                            imported from test.support
? setuptools_scm                      imported from tqdm.version
? simplejson                          imported from requests.compat
? socks                               imported from urllib3.contrib.socks
? urllib.getproxies                   imported from requests.compat
? urllib.getproxies_environment       imported from requests.compat
? urllib.proxy_bypass                 imported from requests.compat
? urllib.proxy_bypass_environment     imported from requests.compat
? urllib.quote                        imported from requests.compat
? urllib.quote_plus                   imported from requests.compat
? urllib.unquote                      imported from requests.compat
? urllib.unquote_plus                 imported from requests.compat
? urllib.urlencode                    imported from requests.compat
? urllib2                             imported from requests.compat
? urlparse                            imported from requests.compat
? util.SKIPPABLE_HEADERS              imported from urllib3.connection
? util.SKIP_HEADER                    imported from urllib3.connection
Building 'dist\main.exe'.
Building shared code archive 'dist\library.zip'.
Copy c:\program files\python39\python39.dll to dist
Copy C:\Program Files\Python39\DLLs\unicodedata.pyd to dist\unicodedata.pyd
Copy C:\Program Files\Python39\DLLs\_bz2.pyd to dist\_bz2.pyd
Copy C:\Program Files\Python39\DLLs\_lzma.pyd to dist\_lzma.pyd
Copy C:\Program Files\Python39\DLLs\_hashlib.pyd to dist\_hashlib.pyd
Copy C:\Program Files\Python39\DLLs\select.pyd to dist\select.pyd
Copy C:\Program Files\Python39\DLLs\pyexpat.pyd to dist\pyexpat.pyd
Copy C:\Program Files\Python39\DLLs\_socket.pyd to dist\_socket.pyd
Copy C:\Program Files\Python39\DLLs\_ssl.pyd to dist\_ssl.pyd
Copy C:\Program Files\Python39\DLLs\_queue.pyd to dist\_queue.pyd
Copy C:\Program Files\Python39\DLLs\_elementtree.pyd to dist\_elementtree.pyd
Copy C:\Program Files\Python39\DLLs\_ctypes.pyd to dist\_ctypes.pyd
Copy C:\Program Files\Python39\DLLs\_tkinter.pyd to dist\_tkinter.pyd
Copy C:\Program Files\Python39\DLLs\_testinternalcapi.pyd to dist\_testinternalcapi.pyd
Copy C:\Program Files\Python39\DLLs\_testcapi.pyd to dist\_testcapi.pyd
Copy C:\Program Files\Python39\DLLs\_asyncio.pyd to dist\_asyncio.pyd
Copy C:\Program Files\Python39\DLLs\_decimal.pyd to dist\_decimal.pyd
Copy C:\Program Files\Python39\DLLs\_multiprocessing.pyd to dist\_multiprocessing.pyd
Copy C:\Program Files\Python39\DLLs\_overlapped.pyd to dist\_overlapped.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\win32gui.pyd to dist\win32gui.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\win32process.pyd to dist\win32process.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\_win32sysloader.pyd to dist\_win32sysloader.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\win32api.pyd to dist\win32api.pyd
Copy C:\Program Files\Python39\DLLs\_uuid.pyd to dist\_uuid.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\yaml\_yaml.cp39-win_amd64.pyd to dist\yaml._yaml.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32comext\shell\shell.pyd to dist\win32com.shell.shell.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\win32console.pyd to dist\win32console.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imaging.cp39-win_amd64.pyd to dist\PIL._imaging.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\win32\win32event.pyd to dist\win32event.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imagingcms.cp39-win_amd64.pyd to dist\PIL._imagingcms.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imagingft.cp39-win_amd64.pyd to dist\PIL._imagingft.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imagingmath.cp39-win_amd64.pyd to dist\PIL._imagingmath.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imagingmorph.cp39-win_amd64.pyd to dist\PIL._imagingmorph.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_imagingtk.cp39-win_amd64.pyd to dist\PIL._imagingtk.pyd
Copy C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\PIL\_webp.cp39-win_amd64.pyd to dist\PIL._webp.pyd
Copy DLL C:\Program Files\Python39\DLLs\tk86t.dll to dist\
Copy DLL C:\Program Files\Python39\DLLs\libssl-1_1.dll to dist\
Copy DLL C:\Program Files\Python39\DLLs\tcl86t.dll to dist\
Copy DLL C:\Program Files\Python39\DLLs\libffi-7.dll to dist\
Copy DLL C:\Program Files\Python39\DLLs\libcrypto-1_1.dll to dist\
Copy lib file C:\Users\MyUser\PycharmProjects\MyProject\venv\lib\site-packages\certifi\cacert.pem to dist\
Copy ExtensionDLL C:\WINDOWS\SYSTEM32\pythoncom39.dll to dist\
Copy ExtensionDLL C:\WINDOWS\SYSTEM32\pywintypes39.dll to dist\

Process finished with exit code 0
```

Is it a bug, or something new and undocumented?
Haven't seen any changes in docs yet :/
![image](https://user-images.githubusercontent.com/27343275/148029180-449995d7-5dd4-4c27-99e2-0361ea700581.png)

**Comment by albertosottile** [comment]
Hi, thanks for reporting this, I will investigate it further. Would it be possible for you to provide a minimal example (application module and `setup.py` file) that reproduces the issue?

I am afraid I also have to point out that the documentation on the website is extremely outdated. Unfortunately, I do not have the bandwidth to take care of that as well (see #53).

**Comment by MaxBQb** [comment]
After some test I find out next things:

Firstly, you were right, it's only appears when `pystray` are in use:
Python 3.10
Windows 10

`pip freeze` = 
```
cachetools==5.0.0
future==0.18.2
pefile==2021.9.3
Pillow==9.0.0
py2exe==0.11.0.1
pystray==0.19.2
six==1.16.0
```

main.py contains nothing but one import:
```py
import pystray
```

setup.py:
```
from distutils.core import setup
import py2exe


setup(console=[dict(
        script='main.py',
      )],
      name="example",
      version="0.0.1",
      options=dict(
          py2exe=dict(
              optimize=1,
          )
      ))
```
Error output ends with:
```
  File "D:\scripts\issueReproduction\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 555, in find_module
    return _find_module(name, path)
  File "D:\scripts\issueReproduction\venv\lib\site-packages\py2exe\vendor\modulefinder.py", line 89, in _find_module
    spec = importlib.machinery.PathFinder.find_spec(name, path)
  File "<frozen importlib._bootstrap_external>", line 1430, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1396, in _get_spec
TypeError: 'int' object is not iterable

Process finished with exit code 1
```

When optimize=0 build works like charm

**Comment by albertosottile** [comment]
This looks like a bug. I attempted a fix in the `develop` branch. Could you download preliminary wheels from Actions here:
https://github.com/py2exe/py2exe/actions/runs/1674893214
and check if those fix this bug also for you? Thank you.

**Comment by MaxBQb** [comment]
It works!
For both test and real project.

**Comment by albertosottile** [comment]
Fixed in 0.11.1.0. Thanks again for having reported this.
